docs/src/components/Layout/Header.tsx

```diff
@@ -79,6 +84,11 @@ const Nav = (props) => (
 
 const Settings = ({ platform, setColorMode, colorMode }) => (
   <Flex className="docs-settings" justifyContent="center" alignItems="center">
+    <DocSearch
+      appId={process.env.DOCSEARCH_DOCS_APP_ID}
```

Comment:
When we have our docs site switch back to SSR? I was under the impression that it is problematic with redirect back then so I switch it back to static site

Reply:
Did you mean the `process.env` was not available for SSR? 

I'm not sure if and when we'll switch to SSR

Reply:
`prcoess.env` only available in SSR

Reply:
Even if this works in SSR mode, I don't understand how these `process.env` variables will be available when React hydrates this code.

Docs seem to indicate you have to make them public: https://nextjs.org/docs/basic-features/environment-variables#exposing-environment-variables-to-the-browser

Do you know how this is working?

Reply:
Yeah, but I am not super confident that I have the entire context. It seems weird to reference `process.env` in component code though

Reply:
- [ ] no new console errors
- [ ] change to use [next public expose iD ](https://nextjs.org/docs/basic-features/environment-variables#exposing-environment-variables-to-the-browser)

Reply:
The console error was not resolved after updating the variables' name

![image](https://user-images.githubusercontent.com/11983489/170607419-311df293-5dec-4ed0-9c20-ce388a5ca7f7.png)

Reply:
We actually don't need to add `NEXT_PUBLIC_` because we did it in docs/next.config.js above [ref](https://nextjs.org/docs/api-reference/next.config.js/environment-variables)

so I'll revert the change

Reply:
I did a quick test creating a new NextJS app and add [Docsearch](https://docsearch.algolia.com/docs/DocSearch-v3) and got the same error 😂

Reply:
Fixed the console error by adding `useEffect`